assembl/auth/generic_auth_backend.py

```diff
@@ -0,0 +1,152 @@
+import base64
+
+from social_core.backends.utils import (
+    get_backend, load_backends as social_load_backends, module_member)
+from social_core.backends.base import BaseAuth
+from social_core.backends.oauth import BaseOAuth2
+import social_pyramid.utils
+import jsonpath_ng
+from six import string_types
+
+from ..lib.raven_client import capture_message
+
+
+class GenericAuth(BaseAuth):
+    """A backend which takes all its attributes from configuration file"""
+    def __init__(self, strategy=None, redirect_uri=None, name=None):
+        self.name = name
+        super(GenericAuth, self).__init__(strategy, redirect_uri)
+        data = self.setting('GENERICAUTH_SUBCONFIGS')
+        assert name in data
+        # override class variables with local
+        self.__dict__.update(data[name])
+
+    def setting(self, name, default=None):
+        """Look in data first"""
+        if name in self.__dict__:
+            return getattr(self, name)
+        return super(GenericAuth, self).setting(name, default)
+
+    @staticmethod
+    def extract_path(data, path):
+        r = jsonpath_ng.parse(path).find(data)
+        if r:
+            return r[0].value
+        capture_message("Could not find %s in %s" % (path, data))
+
+    def format_json_vals(self, json, data):
+        return {k: v.format(**data) if isinstance(v, string_types) else v
+                for (k, v) in json.items()}
+
+    def getResponseInfo(self, prop_name, response):
+        # Simple case
+        rPropName = self.setting(prop_name.upper() + "_PROP", None)
+        if rPropName:
+            return response.get(rPropName, None)
+        # Complex case: use paths
+        rPropName = self.setting(prop_name.upper() + "_PATH", None)
+        if rPropName:
+            return self.extract_path(response, rPropName)
+
+
+class GenericOAuth2(GenericAuth, BaseOAuth2):
+    """A Generic OAuth2 client"""
+
+    # properties and settings from BaseAuth
+    # EXTRA_DATA: None
+    # REQUIRES_EMAIL_VALIDATION: False
+    # SEND_USER_AGENT: False
+    # SSL_PROTOCOL: None
```

Comment:
Safe to send server-to-server request without SSL ?

Reply:
Most of the time it will be SSL. Those are just the defaults extracted from  the social_core code.